### DISCUSSION TITLE
Using outside types in Rune VM (Decimal numbers)

### DISCUSSION THREAD

**Discussion by physics515** [rune-rs/rune#579] [General]
Hello All,

Is it possible, if so, are there any good examples of using outside (rust) crates/types in Rune?

For example, I am experimenting with creating a spreadsheet app, where each cell on the spreadsheet can contain a Rune script (vs formulas in excel). However I'd like to use the BigDecimal ([https://docs.rs/bigdecimal/latest/bigdecimal/](https://docs.rs/bigdecimal/latest/bigdecimal/)) type for numeric calculations. How do I send the BigDecimal crate to the VM context?... if that make sense.

Any help would be much appreciated.

Thank you,

**Comment by udoprog** [comment]
Hi!

One of the modules in [rune-modules](https://github.com/rune-rs/rune/blob/main/crates/rune-modules/src/http.rs) should give you a rough idea of how this is done.

You basically have to make new types out of everything you export into Rune.
